overlord/devicestate/devicestate.go

```diff
@@ -202,11 +203,63 @@ func checkGadgetOrKernel(st *state.State, snapInfo, curInfo *snap.Info, _ snap.C
 	return nil
 }
 
+var (
+	// TODO: replace with gadget.IsCompatible() once we are ready
+	gadgetIsCompatible = func(current, new *gadget.Info) error { return nil }
+)
+
+func checkGadgetRemodelCompatible(st *state.State, snapInfo, curInfo *snap.Info, snapf snap.Container, flags snapstate.Flags, deviceCtx snapstate.DeviceContext) error {
```

Comment:
Maybe a comment why we run it here instead of checking early? iirc the reason is that we need the mounted gadget snap to access the meta/gadget.yaml.